### ISSUE TITLE
TextDecoder is not defined when running `email dev`

### ISSUE THREAD

**Issue by ffiona** [resend/react-email#1258]
### Describe the Bug

I'm not sure if it's a bug or my specific usage of the react-emal, but everything worked as expected before `"react-email": "2.0.0"`. Last working version is 1.9.1.

I want to see pre-rendered emails in the preview so my `export default function` is an output of the `render` utility. 

<img width="987" alt="Screenshot 2024-01-26 at 15 39 15" src="https://github.com/resend/react-email/assets/1404938/46df8162-47fe-4ec0-9bdd-ce7e5ae10025">

### Which package is affected (leave empty if unsure)

react-email

### Link to the code that reproduces this issue

https://github.com/resend/react-email

### To Reproduce

Here is an example of my email:

```
import { Text, Tailwind, Head, Body, Html, render } from '@react-email/components';
import * as React from 'react';
import { CustomTailwindConfig } from './type_system/styles/tailwind.config';

export function Template(props) {
  const { cardText } = props;

  return (
    <Tailwind
      config={CustomTailwindConfig}>
      <Head />
      <Body
        className="bgColor-body">
        <Text>
          {cardText}
        </Text>
      </Body>
    </Tailwind>
  );
}

export function Email() {

  const emailHtml = render(
    <Template
      cardText="Test"
    />, {
      pretty: false,
    }
  );

  const doctype = '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">';
  const rawEmail = emailHtml.replace(doctype,'');

  return (
    <Html dangerouslySetInnerHTML={{ __html: rawEmail }}></Html>
  );
}

export default Email;

```

### Expected Behavior

Just a working desktop & source preview:

<img width="497" alt="Screenshot 2024-01-26 at 17 09 49" src="https://github.com/resend/react-email/assets/1404938/4aea4e34-ff1a-4296-b7d0-6f1c7a3ca44d">
<img width="778" alt="Screenshot 2024-01-26 at 17 10 14" src="https://github.com/resend/react-email/assets/1404938/5be29d07-6fa7-4a24-8a8d-335986627c2b">

### What's your node version? (if relevant)

18.17.0

**Comment by gabrielmfern** [comment]
This is indeed a bug here, thank you for opening this issue.

---

On another note, you might not need to the what you mentioned you did,

> I want to see pre-rendered emails in the preview so my export default function is an output of the render utility.

this is because we already handle this and use `renderAsync` under the hood to get the actual email HTML into the iframe for you. 

You can see this from the source code:

https://github.com/resend/react-email/blob/0d03f1cf5c5d1d23c1e93fb558c2b6205fefe2dc/packages/react-email/src/actions/render-email-by-slug.tsx#L35-L45

**Comment by ffiona** [comment]
Thank you for the quick reply!

> If you remove this render you mention you won't face this issue anymore.

Yeah, I've checked it, and everything was working fine, that's why I assumed that this exact case with `render` can be my own specific case. But the reason I build emails this way is an easier email testing with different data. I have a flag `isDevMode` and setting it `true` or `false` I can see a rendered email with the text or with some special syntax for our backend.

Let me know if using `render` utility this way is incorrect, so I could think about any other solution for my purposes :sweat_smile:

**Comment by gabrielmfern** [comment]
> But the reason I build emails this way is an easier email testing (and exporting via npm run export) with different data. I have a flag isDevMode and setting it true or false I can see a rendered email with the text or with some special syntax for our backend.

Oh, I see, then you'll have to wait a bit for us to release the PR for this 😆, I've just opened it #1259 though.

I suppose having to do this kind of thing isn't the best experience for you. 
I've got an idea that might be helpful for you if you don't use special syntax like for loops.
What you can do is, instead of directly adding the templating syntax on the email, you receive them as props to the email component, and then you can define the default values for these props using the templating syntax. After that, you can define your `PreviewProps` to be what you expect the values to be like.

This can also be a foot gun for others who come around this thinking these values might be dynamic as they won't when rendering the email outside of the preview.

**Comment by ffiona** [comment]
@gabrielmfern thank you! ☺️

**Comment by luismramirezr** [comment]
I fixed it by adding this to my Jest "setupFiles" file:
```typescript
import { TextEncoder, TextDecoder } from 'util';

Object.assign(global, { TextDecoder, TextEncoder });
```

Now I'm just getting some warnings:
```  console.error
    Warning: validateDOMNesting(...): <div> cannot appear as a child of <html>.
        at div```

**Comment by gabrielmfern** [comment]
@luismramirezr Seems like something to do with the version of jsdom there https://stackoverflow.com/questions/68468203/why-am-i-getting-textencoder-is-not-defined-in-jest 🤔

I'll hide your comments just to avoid confusion for anyone coming into this issue as well

**Comment by ffiona** [comment]
@gabrielmfern any news if the fix could be reviewed?

**Comment by gabrielmfern** [comment]
> @gabrielmfern any news if the fix could be reviewed?

Just did! Going to make a canary release right now.

**Comment by gabrielmfern** [comment]
Released on `react-email@2.1.1-canary.0`

**Comment by ffiona** [comment]
@gabrielmfern everything works as expected now, thanks alot! 👍
